src/Access/AccessControlManager.cpp

```diff
@@ -79,6 +80,12 @@ void AccessControlManager::setLocalDirectory(const String & directory_path)
 }
 
 
+void AccessControlManager::setExternalAuthenticatorsConfig(const Poco::Util::AbstractConfiguration & config)
+{
+    external_authenticators = std::make_unique<ExternalAuthenticators>(config, getLogger());
```

Comment:
I think the code can be easier (and safer) if you create `external_authenticators` in the constructor of `AccessControlManager` and just call `external_authenticators->setConfig(config)` here.

Reply:
I am not a big fan of all these setters and getters. They results in a code whose state transformation are impossible to track. I'll modify the code as you suggest, but it won't make it better, just as bad as it is now.

Reply:
Pushed the change, please take a look.